Systems/WorldGen/Standard/ChunkGen/1.GenTerra/GenTerra.cs

```diff
@@ -314,15 +307,20 @@ private void OnChunkColumnGen(IServerChunk[] chunks, int chunkX, int chunkZ, ITr
             int taperThreshold = (int)(mapsizeY * 0.9f);
             double geoUpheavalAmplitude = 255;
 
-            for (int dx = 0; dx <= 1; dx++)
+            for (int xN = 0; xN < paddedNoiseWidth; xN++)
             {
-                for (int dz = 0; dz <= 1; dz++)
+                for (int zN = 0; zN < paddedNoiseWidth; zN++)
                 {
-                    VectorXZ distGeo = DistortionNoise((chunkX + dx) * chunksize, (chunkZ + dz) * chunksize);
+                    VectorXZ distGeo = DistortionNoise(chunkX * chunksize + xN * lerpHor, chunkZ * chunksize + zN * lerpHor);
                     distGeo = ApplyDistortionThreshold(distGeo * 10.0, 10.0);
 
-                    distxz[dx, dz][0] = distGeo.X;
-                    distxz[dx, dz][1] = distGeo.Z;
+                    float upheavalStrength = GameMath.BiLerp(upheavalMapUpLeft, upheavalMapUpRight, upheavalMapBotLeft, upheavalMapBotRight,
+                        xN * (1.0f / noiseWidth), zN * (1.0f / noiseWidth));
+                    float distYHere = ComputeOceanAndUpheavalDistY(upheavalStrength,
+                                chunkX * chunksize + xN * lerpHor, chunkZ * chunksize + zN * lerpHor,
+                                distGeo);
+
+                    distY[NoiseIndex2d(xN, zN)] = distYHere;

```

Comment:
Previously generated the XZ displacements at chunk corners and upheaval every trilerp column. Changed this to computeXZ displacements and upheaval for every trilerp column, in one go. 2D noise is almost never a hot codepath in a 3D noise generator, so you might as well get the most you can out of it.

No longer storing the X and Z displacements in the array, because they're used to calculate the upheaval right away. The array now covers every trilerp column in the chunk.